src/pages/index.js

```diff
@@ -0,0 +1,290 @@
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  cardImageEl.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalCaptionEl.textContent = data.name;
+    previewModalImageEl.src = data.link;
+    previewModalImageEl.alt = data.name;
+  });
+
+  if (data.isLiked) {
+    cardLikeBtn.classList.add("card__like-btn_liked");
+  }
+
+  cardLikeBtn.addEventListener("click", (evt) => handleLike(evt, data._id));
+
+  cardDeleteBtn.addEventListener("click", () =>
+    handleDeleteCard(cardElement, data._id)
+  );
+
+  return cardElement;
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keyup", closeModalOnEsc);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keyup", closeModalOnEsc);
+}
+
+function closeModalOnEsc(evt) {
+  if (evt.key === "Escape") {
+    const activeModal = document.querySelector(".modal_opened");
+    closeModal(activeModal);
+  }
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  setButtonText(editSubmitBtn, true, "Save", "Saving...");
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(editSubmitBtn, false, "Save", "Saving...");
+    });
+}
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+  setButtonText(cardSubmitBtn, true, "Save", "Saving...");
+  const inputValues = {
+    name: cardNameInput.value,
+    link: cardLinkInput.value,
+  };
+  api
+    .addNewCard(inputValues)
+    .then((data) => {
+      const cardElement = getCardElement(data);
+      cardsList.prepend(cardElement);
+      evt.target.reset();
+      closeModal(cardModal);
+      disableButton(cardSubmitBtn, settings);
+      console.log(cardSubmitBtn);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(cardSubmitBtn, false, "Save", "Saving...");
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  setButtonText(avatarSubmitBtn, true, "Save", "Saving...");
+
+  const avatarInput = document.getElementById("profile-avatar-input");
+
+  api
+    .editAvatarInfo(avatarInput.value)
+    .then((data) => {
+      document.getElementById("image-avatar").src = data.avatar;
+      closeModal(avatarModal);
+      avatarLinkInput.value = "";
+      disableButton(avatarSubmitBtn, settings);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(avatarSubmitBtn, true, "Save", "Saving...");
+    });
+}
+
+profileEditButton.addEventListener("click", () => {
+  editModalNameInput.value = profileName.textContent;
+  editModalDescriptionInput.value = profileDescription.textContent;
+  disableButton(editSubmitBtn, settings);
```

Comment:
Please, call `resetValidation` when you open the profile modal (it was according to the 6th sprint). You shouldn't change the code of that sprint for the profile modal handling


```js

  resetValidation(editFormElement, settings);

```